### PR TITLE
feat: leverage `web.list` pagination

### PR DESCRIPTION
**Pull request by fmorency**
- Better loading and error handling
- Simplify queries

Contains #38

### CODE REVIEW

**Review by jgryffindor — APPROVED** [review]
Tested and looks good!

File: src/features/deployments/queries.ts [hunk 1]
```diff
@@ -109,87 +109,81 @@ interface DeploymentsListArgs {
...
+  >(['web', 'list', address, page], fetchDeployments, {
+    keepPreviousData: true,
+  })
 
-  const total = data.length
+  const total = data?.totalCount ?? 0
   const numPages = Math.ceil(total / PAGE_SIZE)
-  const hasNextPage = currentPage < numPages
+  const hasNextPage = page < numPages
+  const deployments = data?.deployments ?? []
```
Comment (nblenke) [inline src/features/deployments/queries.ts:161]: In testing I am getting the same set of deployments on page 2 as I do on page 1. It's also possible that I may be missing something here

Comment (fmorency) [reply src/features/deployments/queries.ts:161]: You need to test against the `many-rs` branch found in https://github.com/liftedinit/many-rs/pull/414

Comment (nblenke) [reply src/features/deployments/queries.ts:161]: I see. I was able to test and it looks great!

**Review by nblenke — APPROVED** [review]

**Pull request merged**